testing/unittests/jthreadpooltests.cpp

```diff
@@ -0,0 +1,943 @@
+    {
+        // Create thread pool with max 2 threads and a default start delay of 1 second.
+        // Testing thread start() with and without start delay
+        pool.setown(createThreadPool("ThrottledThreadPoolWithDefaultDelay", factory, true, nullptr, 2));
+
+        // Both threads should start immediately
+        params = {lifespan500milliseconds, "Thread1"};
+        auto beforeStart = std::chrono::high_resolution_clock::now();
+        pool->start(&params, params.name);
+        params = {lifespan500milliseconds, "Thread2"};
+        pool->start(&params, params.name);
+
+        // Wait for both threads to signal that they've started
+        auto duration = measureThreadStartDuration(beforeStart, 2);
+        CPPUNIT_ASSERT(duration.count() < 1000);
+        CPPUNIT_ASSERT_EQUAL(2U, threadStartedCount.load());
+
+        // The Thread3 should not start after waiting on it's start specified timeout as the other
+        // two threads are still running when the timeout expires
+        params = {lifespan500milliseconds, "Thread3"};
+        unsigned thread3ExceptionCount = 0;
+        attemptThreadStartWithTimeout(&params, milliseconds200, "ThrottledThreadPoolWithDefaultDelay", thread3ExceptionCount,
+                                      true, "Start should have thrown an exception as no slots would be available within thread start delay specified");
+        CPPUNIT_ASSERT_EQUAL(2U, threadStartedCount.load());
+
+        // The Thread4 should start after one of the above completes
+        beforeStart = std::chrono::high_resolution_clock::now();
+        params = {lifespan750milliseconds, "Thread4"};
+        unsigned thread4ExceptionCount = 0;
+        attemptThreadStartWithTimeout(&params, milliseconds1000, "ThrottledThreadPoolWithDefaultDelay", thread4ExceptionCount, false);
+        duration = measureThreadStartDuration(beforeStart, 1);
+        CPPUNIT_ASSERT(duration.count() >= 290);
+        CPPUNIT_ASSERT_EQUAL(3U, threadStartedCount.load());
+        CPPUNIT_ASSERT(threadCompletedCount.load() >= 1);
+        CPPUNIT_ASSERT(pool->runningCount() >= 1);
+
+        // Test Thread Pool default start delay
+        //
+        beforeStart = std::chrono::high_resolution_clock::now();
+        params = {lifespan1000milliseconds, "Thread5"};
+        pool->start(&params, params.name);
+        duration = measureThreadStartDuration(beforeStart, 1);
+        CPPUNIT_ASSERT(duration.count() < 100);
+        CPPUNIT_ASSERT(pool->runningCount() >= 2);
+
+        // Thread6 should start up to the Thread Pool default start delay of 1 second
+        params = {lifespan1000milliseconds, "Thread6"};
+        beforeStart = std::chrono::high_resolution_clock::now();
+        pool->start(&params, params.name); // no exception should be thrown as no timeout is specified on the start
+        duration = measureThreadStartDuration(beforeStart, 1);
+        CPPUNIT_ASSERT(duration.count() >= 749 && duration.count() < 1000);
```

Comment:
could be less than 749 couldn't it? needs an comment explaining logic of expectation.